kubernetes/vault-snapshot.sh

```diff
@@ -7,8 +7,22 @@ VAULT_TOKEN=$(vault write -field=token  auth/kubernetes/login role="${VAULT_ROLE
 export VAULT_TOKEN
 
 # create snapshot
-
 vault operator raft snapshot save /vault-snapshots/vault_"$(date +%F-%H%M)".snapshot
 
 # upload to s3
 s3cmd put /vault-snapshots/* "${S3_URI}" --host="${S3_HOST}" --host-bucket="${S3_BUCKET}"
+
+# remove expired snapshots
+if [ "${S3_EXPIRE_DAYS}" ]; then
+    s3cmd ls "${S3_URI}" --host="${S3_HOST}" --host-bucket="${S3_BUCKET}" | while read -r line; do
+        createDate=$(echo $line | awk {'print $1" "$2'})
+        createDate=$(date -d"$createDate" +%s)
+        olderThan=$(date --date "${S3_EXPIRE_DAYS} days ago" +%s)
+        if [[ $createDate -lt $olderThan ]]; then
+            fileName=$(echo $line | awk {'print $4'})
+            if [[ $fileName != "" ]]; then
+                s3cmd del "${S3_URI}/$fileName" --host="${S3_HOST}" --host-bucket="${S3_BUCKET}"
```

Comment:
Also, there seems to be quite some inconsistency between the delete behavior of the client tools.

* `s3cmd del` does exactly nothing when running against a file that is locked
* `mc rm` actually puts the `DEL` marker

Definitely needs to be tested with a real case and a low `$S3_EXPIRE_DAYS`

Reply:
It could easily be that not all command line tool handle the full S3 API. In this case we would probably have to run the commands verbose to check which API calls both tools are actually making.

Reply:
I had a bug in my shell script. The `$fileName` already contains the entire S3 URI. Now that this is fixed, it actually deletes the remote objects when `S3_EXPIRE_DAYS` is defined!